src/main/java/com/sofrelgroup14/ci/controller/BuildController.java

```diff
@@ -1,34 +1,130 @@
+     * 'git clone -b webhooks-playground <repo-url> repo'
+     * 'mvn verify --file repo/pom.xml'
+     * ...save the output to the database...
+     * 'rm -Rf repo'
+     *
+     * What HTTP-payload looks like can be seen in Settings -> Webhooks
+     * on GitHub (click an active webhook and scroll down).
+     *
+     * @param body The HTTP POST request body.
+     */
     @RequestMapping(value = "", method = RequestMethod.POST)
-    public Logs buildAndCreateLog() {
-        Logs log = new Logs(ObjectId.get(), true, "great job", Instant.now());
-        repository.save(log);
-        return log;
+    public void buildAndCreateLog(@RequestBody String body) {
 
-    }
+        // TODO (notification): Set commit status to 'pending'
+
+        String ref;
+        String commitHash = ""; // Remove error: "commitHash may not have been initialized"
+        String branchName = ""; // -||-
+
+        // Parse the JSON in the POST body to extract commit-hash and branch name.
+        // (Source for JSON parsing strings automatically: https://stackoverflow.com/a/5245881)
+        try {
+            JSONObject jsonObject = new JSONObject(body);
+
+            ref = jsonObject.getString("ref"); // Will be "refs/heads/branch-name"
+            String[] tmp = ref.split("/");
+            branchName = tmp[tmp.length-1];
+
+            commitHash = jsonObject.getString("after");
+
+        } catch (JSONException e){
+            e.printStackTrace();
+        }
 
+        // Clone repo, run mvn verify, post output to db, remove repo
+        // ProcessBuilder:s have to be wrapped in try-catch block.
+        try {
+            ProcessBuilder pbClone = new ProcessBuilder(
+              "git", "clone", "-b", branchName, "https://github.com/sofrel-group14/ci.git", "repo"); // Could be useful to save time and space: '--single-branch' ("Clone only the history leading to the tip of a single branch, either specified by the --branch option or the primary branch remote's HEAD points at.")
+            pbClone.start().waitFor(); // Start process and block this program thread until process has finished.
+
+            ProcessBuilder pbMvnVerify = new ProcessBuilder(
+              "mvn", "verify", "--file", "repo/pom.xml");
+            Process p = pbMvnVerify.start();
+
+            // Gather Maven-output (i.e. build output) to string (source: https://stackoverflow.com/a/16714180)
+            BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String line;
+            StringBuilder sb = new StringBuilder();
+            while ((line = reader.readLine()) != null) {
+                sb.append(line).append("\n");
+            }
+            String buildOutput = sb.toString();
+
+            // See if build was successful
+            p.waitFor(); // Wait until process is finished
+            int val = p.exitValue(); // 0 if success, error code otherwise
+            boolean buildSuccess = (val == 0) ? true : false;
+
```

Comment:
You could probably use the buildSuccess error code to extract more useful data when it fails (for example, if it fails due to compilation error vs tests failing), that could be something to look into if there's time